packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,157 @@
+import userEvent from '@testing-library/user-event';
+
+import { BaseInput } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import ThemeWrapper from '~src/_helpers/testing/themeWrapper';
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+import { CloseIcon, EyeIcon } from '~components/Icons';
+
+describe('<BaseInput />', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(<BaseInput label="Enter name" id="name" />);
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render success validation state', () => {
+    const { container } = renderWithTheme(
+      <BaseInput label="Enter name" id="name" validationState="success" successText="Success" />,
+    );
+
+    expect(container).toMatchSnapshot();
```

Comment:
shouldn't we also be checking that the success text is being displayed? snapshot will help us account for any changes in the layout but whether it's rendering in the first place or not won't be captured using snapshot right?

Reply:
Actually such tests would be much better over snapshots, will swap this out. Personally snapshots are very hard to maintain and review.

Reply:
Done in https://github.com/razorpay/blade/pull/695/commits/1b30d066d0edf03c62601f0c8a2dbb64fe5b7b31